pgsqltoolsservice/query_execution/query_execution_service.py

```diff
@@ -376,9 +373,9 @@ def _get_query_text_from_execute_params(self, params: ExecuteRequestParamsBase):
             # Then params must be an instance of ExecuteStringParams, which has the query as an attribute
             return params.query
 
-    def _resolve_query_exception(self, e: Exception, query: Query, request_context: RequestContext, conn: 'psycopg2.connection', is_rollback_error=False):
+    def _resolve_query_exception(self, e: Exception, query: Query, request_context: RequestContext, conn: ServerConnection, is_rollback_error=False):
         utils.log.log_debug(self._service_provider.logger, f'Query execution failed for following query: {query.query_text}\n {e}')
-        if isinstance(e, psycopg2.DatabaseError) or isinstance(e, RuntimeError) or isinstance(e, psycopg2.extensions.QueryCanceledError):
+        if isinstance(e, conn.database_error) or isinstance(e, RuntimeError):
```

Comment:
database_error is an instance and not the type. I dont think this would work

Reply:
conn.database_error is meant to return the type of DatabaseError this driver throws. I'll be sure to test it again!